.github/workflows/pr.yaml

```diff
@@ -0,0 +1,60 @@
+name: Validate PR
+on:
+  pull_request:
+    branches: [ main ]
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: './go.mod'
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          args: --timeout=5m
+
+  security_scan:
+    name: Security scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: './go.mod'
+
+      # https://github.com/securego/gosec/blob/12be14859bc7d4b956b71bef0b443694aa519d8a/README.md#integrating-with-code-scanning
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
+        with:
+          # we let the report trigger content trigger a failure using the GitHub Security features.
+          args: '-no-fail -fmt sarif -out results.sarif ./...'
+      - name: Upload SARIF file
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          # Path to SARIF file relative to the root of the repository
+          sarif_file: results.sarif
+
+  code_coverage:
+    name: Code Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: './go.mod'
+      - name: Run tests and collect coverage
+        run: make test
+      - name: Upload codecov report
+        uses: codecov/codecov-action@v3.1.1
+        with:
+          fail_ci_if_error: true
```

Comment:
Would it make sense to split this into two jobs like [here](https://github.com/redhat-appstudio/build-service/blob/6588316a6842af5276ab13650702f37d8b7c2568/.github/workflows/pr.yaml)?

Might not be a problem anymore, but I previously encountered problems with uploads to codecov.
If the codecov upload fails, it would require rerunning the entire job, which be time-consuming if `make test` takes several minutes. Rerunning codecov upload takes max 10 seconds.
The split might also simplify the debugging process since we would know if the error occured in the tests or codecov

Reply:
`I previously encountered problems with uploads to codecov` could you please specify what was the exact problem that you encountered? I don't think splinting into two separate jobs one for running the test and another for codecov upload is needed. In case go test step fails, we can debug that from the github actions logs itself. I think the upload artifacts and download artifacts in the build-service github actions overkill.

Reply:
WDYT?

Reply:
> could you please specify what was the exact problem that you encountered?

I wasn't able to find it. Anyway I took a look at some recent logs and the upload doesn't seem to fail anymore, so not an issue.

I believe you're right, debugging from logs is fine, so probably no reason for the split.